wlauto/workloads/skypeecho/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,207 @@
+    public static String voiceCallButtonResourceId = PACKAGE_ID + "chat_menu_item_call_voice";
+    public static String videoCallButtonResourceId = PACKAGE_ID + "chat_menu_item_call_video";
+    public static String endCallButtonResourceId = PACKAGE_ID + "call_end_button";
+    public static String noContactMessage = "Could not find contact \"%s\" in the contacts list.";
+
+    private Map<String, Timer> results = new TreeMap<String, Timer>();
+    private boolean dumpsysEnabled;
+    private String outputDir;
+
+    private static Arguments args;
+
+    private static final class Arguments {
+        String loginName;
+        String loginPass;
+        String contactSkypeid;
+        String contactName;
+        int callDuration;
+        String callType;
+        String resultsFile;
+        String outputDir;
+        boolean dumpsysEnabled;
+    }
+
+    private static Arguments parseBundle(Bundle bundle) {
+        Arguments args = new Arguments();
+        args.loginName = bundle.getString("my_id");
+        args.loginPass = bundle.getString("my_pwd");
+        args.contactSkypeid = bundle.getString("skypeid");
+        args.contactName = bundle.getString("name").replace("_", " ");
+        args.callDuration = Integer.parseInt(bundle.getString("duration"));
+        args.callType = bundle.getString("action");
+        args.resultsFile = bundle.getString("results_file");
+        args.outputDir = bundle.getString("output_dir");
+        args.dumpsysEnabled = Boolean.parseBoolean(bundle.getString("dumpsys_enabled"));
+        return args;
+    }
+
+    public void runUiAutomation() throws Exception {
+        // Override superclass value
+        this.waitTimeout = 10000;
+
+        // Get Params
+        Bundle parameters = getParams();
+        String loginName = parameters.getString("my_id");
+        String loginPass = parameters.getString("my_pwd");
+        String contactSkypeid = parameters.getString("skypeid");
+        String contactName = parameters.getString("name").replace("_", " ");
+        int callDuration = Integer.parseInt(parameters.getString("duration"));
+        String callType = parameters.getString("action");
+        String resultsFile = parameters.getString("results_file");
+        outputDir = parameters.getString("output_dir", "/sdcard/wa-working");
+        dumpsysEnabled = Boolean.parseBoolean(parameters.getString("dumpsys_enabled"));
+
+        // Run tests
+        Timer overallTimer = new Timer();
+        overallTimer.start();
+        handleLoginScreen(loginName, loginPass);
+        selectContact(contactName, contactSkypeid);
+        if ("video".equalsIgnoreCase(callType)) {
+            videoCallTest(callDuration);
+        } else if ("voice".equalsIgnoreCase(callType)) {
+            voiceCallTest(callDuration);
+        }
+        overallTimer.end();
+
+        // Save results
+        results.put("overall_test", overallTimer);
+        saveResults(results, resultsFile);
```

Comment:
Please change this to use writeResultsToFile in the base class so that we write the results in a consistent manner between workloads.  Thanks.